awx/main/credential_plugins/conjur.py

```diff
@@ -68,7 +68,12 @@ def conjur_backend(**kwargs):
     with CertFiles(cacert) as cert:
         # https://www.conjur.org/api.html#authentication-authenticate-post
         auth_kwargs['verify'] = cert
-        resp = requests.post(urljoin(url, '/'.join(['api', 'authn', account, username, 'authenticate'])), **auth_kwargs)
+        try:
+            resp = requests.post(urljoin(url, '/'.join(['authn', account, username, 'authenticate'])), **auth_kwargs)
+        except requests.exceptions.ConnectionError:
```

Comment:
any particular reason to have the redundant exception here and below? should it just be `except:` , or did you have a plan to add to this in the future

Reply:
What this code block is doing is trying to send the API request to the targeted Conjur service first with `/api` provided in the URI.  This will work for self-hosted Conjur Secrets Manager Enterprise and Conjur Cloud.  If a `requests.exceptions.ConnectionError` is returned, this means that a Conjur service does exist, however, it does not support `/api` in the URI.  This would be if Conjur Open Source was being used.  Therefore, we capture the exception and attempt another API request to the Conjur service, this time _without_ the `/api` in the URI.  Any other errors or further errors after the attempt should be handled the same way errors are in Credential Plugins, hence no bare `except`.

Reply:
"first with /api provided in the URI." is it backwards?

seems we first attempt to without `/api`, then with

`'/'.join(['authn', account, username, 'authenticate']`

Reply:
Sorry, you're right. I was looking on mobile and it was skewing my view.  Since API linting is failing on the bare `except`, I'm going to remove that and just keep the `except requests.exceptions.HTTPConnection` handling.

So the code block goes...
1. Try `/authn/` which is supported by Conjur Enterprise and Conjur Open Source.  If HTTPConnection error returned...
2. Try `/api/authn` which is supported by Conjur Cloud.
3. Regardless of success/failure, `raise_for_status` and toast notification in AWX will display results of test, if testing.

I'll make the change now and re-request review.